python-squid/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -518,9 +521,62 @@ private PyExceptClauseTree exceptClause(AstNode except, PyStatementListTree body
   }
 
   PyExpressionTree expression(AstNode astNode) {
+    if (astNode.is(PythonGrammar.TEST) && astNode.getChildren().size() == 1) {
+      return expression(astNode.getFirstChild());
+    }
+    if (astNode.is(PythonGrammar.ATOM) && astNode.getChildren().size() == 1) {
+      return expression(astNode.getFirstChild());
+    }
     if (astNode.is(PythonGrammar.YIELD_EXPR)) {
       return yieldExpression(astNode);
     }
+    if (astNode.is(PythonGrammar.NAME)) {
+      return name(astNode);
+    }
+    if (astNode.is(PythonGrammar.ATTRIBUTE_REF)) {
+      return qualifiedExpression(astNode);
+    }
+    if (astNode.is(PythonGrammar.CALL_EXPR)) {
+      return callExpression(astNode);
+    }
     return new PyExpressionTreeImpl(astNode);
   }
+
+  public PyQualifiedExpressionTree qualifiedExpression(AstNode astNode) {
+    PyExpressionTree qualifier = expression(astNode.getFirstChild());
+    List<AstNode> names = astNode.getChildren(PythonGrammar.NAME);
+    names.remove(names.size() - 1);
+    PyNameTree name = name(astNode.getLastChild());
+    for (AstNode nameNode : names) {
+      qualifier = new PyQualifiedExpressionTreeImpl(astNode, name(nameNode), qualifier);
```

Comment:
I think we can't use `astNode` here, that's not the correct location for the tree. The problem is that there's no instance of `AstNode`. I have the same problem with binary expressions, and I believe the solution is to [stop requiring](https://github.com/SonarSource/sonar-python/pull/271) an `AstNode` to build a `PyTree`.